packages/pds/src/services/actor/index.ts

```diff
@@ -166,6 +166,23 @@ export class ActorService {
     log.info({ handle, email, did }, 'registered user')
   }
 
+  async updateHandle(did: string, handle: string) {
+    const res = await this.db.db
+      .updateTable('did_handle')
+      .set({ handle })
+      .where('did', '=', did)
+      .whereNotExists(
+        this.db.db
+          .selectFrom('did_handle')
+          .where('handle', '=', handle)
+          .selectAll(),
+      )
```

Comment:
We should be able to lean directly on the `handle_lower` index to ensure we don't get a duplicate, which I think would actually feel pretty nice, and prove that we don't strictly rely on trusted `handle` inputs to this method.  That would involve switching this to an `onConflict((oc) => oc.doNothing())`.

Reply:
I would prefer that too & it's what I originally tried, but there's no `onConflict` for `update` :/

Any other ideas?

Reply:
Ah dang, good call!  My only other idea is to handle the constraint error that comes out of the database, but that's a little funky to do across sqlite and pg.  I think as-is works well particularly with a test to back it up 👍

Reply:
cool cool 👍 

got that test in & it works fine